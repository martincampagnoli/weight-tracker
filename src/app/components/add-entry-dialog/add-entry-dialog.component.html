<h2 mat-dialog-title>Add entry</h2>
<mat-dialog-content>
  <app-add-entry [dialogRef]="dialogRef" />
</mat-dialog-content>
