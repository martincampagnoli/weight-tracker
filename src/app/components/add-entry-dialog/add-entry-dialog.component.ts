import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import {
  MatDialogTitle,
  MatDialogContent,
  MatDialogRef,
} from '@angular/material/dialog';
import { AddEntryComponent } from '../add-entry/add-entry.component';

@Component({
  selector: 'add-entry-dialog',
  templateUrl: 'add-entry-dialog.component.html',
  standalone: true,
  imports: [
    MatButtonModule,
    MatDialogTitle,
    MatDialogContent,
    AddEntryComponent,
  ],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AddEntryDialogComponent {
  readonly dialogRef = inject(MatDialogRef<AddEntryDialogComponent>);
}
