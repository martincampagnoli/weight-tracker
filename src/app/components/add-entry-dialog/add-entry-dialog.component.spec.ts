import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatDialogRef } from '@angular/material/dialog';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { provideMockStore } from '@ngrx/store/testing';
import { provideNativeDateAdapter } from '@angular/material/core';
import { AddEntryDialogComponent } from './add-entry-dialog.component';

describe('AddEntryDialogComponent', () => {
  let component: AddEntryDialogComponent;
  let fixture: ComponentFixture<AddEntryDialogComponent>;
  let mockDialogRef: jest.Mocked<MatDialogRef<AddEntryDialogComponent>>;

  const initialState = {
    data: []
  };

  beforeEach(async () => {
    mockDialogRef = {
      close: jest.fn(),
      afterClosed: jest.fn(),
      beforeClosed: jest.fn(),
    } as any;

    await TestBed.configureTestingModule({
      imports: [AddEntryDialogComponent, NoopAnimationsModule],
      providers: [
        provideMockStore({ initialState }),
        provideNativeDateAdapter(),
        {
          provide: MatDialogRef,
          useValue: mockDialogRef
        }
      ]
    }).compileComponents();

    fixture = TestBed.createComponent(AddEntryDialogComponent);
    component = fixture.componentInstance;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Component Creation', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
      fixture.detectChanges();
      expect(fixture).toMatchSnapshot();
    });
  });

  describe('Properties', () => {
    it('should inject MatDialogRef', () => {
      expect(component.dialogRef).toBeDefined();
      expect(component.dialogRef).toBe(mockDialogRef);
    });
  });

  describe('Template Rendering', () => {
    it('should render AddEntryComponent', () => {
      fixture.detectChanges();

      const addEntryComponent =
        fixture.debugElement.nativeElement.querySelector('app-add-entry');
      expect(addEntryComponent).toBeTruthy();
    });
  });
});
