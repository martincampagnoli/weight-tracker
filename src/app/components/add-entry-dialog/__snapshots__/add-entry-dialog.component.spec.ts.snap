// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEntryDialogComponent Component Creation should match snapshot 1`] = `
<add-entry-dialog
  dialogRef={[Function Object]}
>
  <h2
    class="mat-mdc-dialog-title mdc-dialog__title"
    id="mat-mdc-dialog-title-a1"
    mat-dialog-title=""
  >
    Add entry
  </h2><mat-dialog-content
    class="mat-mdc-dialog-content mdc-dialog__content"
  >
    <app-add-entry>
      <mat-stepper
        aria-orientation="vertical"
        class="stepper-form mat-stepper-vertical"
        linear=""
        role="tablist"
        style="--mat-stepper-animation-duration: 0ms;"
      >
        <div
          class="mat-step"
        >
          <mat-step-header
            aria-posinset="1"
            aria-selected="true"
            aria-setsize="4"
            class="mat-step-header mat-vertical-stepper-header mat-primary"
            role="tab"
            tabindex="0"
          >
            <div
              class="mat-ripple mat-step-header-ripple mat-focus-indicator"
              matripple=""
            />
            <div
              class="mat-step-icon mat-step-icon-state-number mat-step-icon-selected"
            >
              <div
                class="mat-step-icon-content"
              >
                <span
                  aria-hidden="true"
                >
                  1
                </span>
              </div>
            </div>
            <div
              class="mat-step-label mat-step-label-active mat-step-label-selected"
            >
              <div
                class="mat-step-text-label"
              >
                Insert Weight
              </div>
            </div>
          </mat-step-header>
          <div
            class="mat-vertical-content-container mat-stepper-vertical-line mat-vertical-content-container-active"
          >
            <div
              class="mat-vertical-stepper-content"
              role="tabpanel"
            >
              <div
                class="mat-vertical-content"
              >
                <form
                  class="ng-untouched ng-pristine ng-invalid"
                  novalidate=""
                >
                  <mat-form-field
                    class="mat-mdc-form-field pe-1 pt-1 weight-input mat-mdc-form-field-type-mat-input mat-form-field-appearance-fill mat-form-field-hide-placeholder mat-primary ng-untouched ng-pristine ng-invalid"
                  >
                    <div
                      class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled"
                    >
                      <div
                        class="mat-mdc-form-field-focus-overlay"
                      />
                      <div
                        class="mat-mdc-form-field-flex"
                      >
                        <div
                          class="mat-mdc-form-field-infix"
                        >
                          <label
                            class="mdc-floating-label mat-mdc-floating-label"
                            for="mat-input-a3"
                            id="mat-mdc-form-field-label-a3"
                            matformfieldfloatinglabel=""
                          >
                            <mat-label>
                              kgs
                            </mat-label>
                            <span
                              aria-hidden="true"
                              class="mat-mdc-form-field-required-marker mdc-floating-label--required"
                            />
                          </label>
                          <input
                            aria-required="true"
                            class="mat-mdc-input-element ng-untouched ng-pristine ng-invalid mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
                            formcontrolname="weight"
                            id="mat-input-a3"
                            matinput=""
                            required=""
                          />
                        </div>
                      </div>
                      <div
                        class="mdc-line-ripple mdc-line-ripple--deactivating"
                        matformfieldlineripple=""
                      />
                    </div>
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
                    >
                      <div
                        class="mat-mdc-form-field-hint-wrapper"
                      >
                        <div
                          class="mat-mdc-form-field-hint-spacer"
                        />
                      </div>
                    </div>
                  </mat-form-field>
                  <br />
                  <div
                    class="button-container"
                  >
                    <button
                      class="mdc-button mat-mdc-button-base mat-stepper-next mdc-button--raised mat-mdc-raised-button mat-unthemed _mat-animation-noopable"
                      mat-button=""
                      mat-raised-button=""
                      mat-ripple-loader-class-name="mat-mdc-button-ripple"
                      mat-ripple-loader-uninitialized=""
                      matsteppernext=""
                      type="submit"
                    >
                      <span
                        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        Next
                      </span>
                      <span
                        class="mat-focus-indicator"
                      />
                      <span
                        class="mat-mdc-button-touch-target"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mat-step"
        >
          <mat-step-header
            aria-disabled="true"
            aria-posinset="2"
            aria-selected="false"
            aria-setsize="4"
            class="mat-step-header mat-vertical-stepper-header mat-primary"
            role="tab"
            tabindex="-1"
          >
            <div
              class="mat-ripple mat-step-header-ripple mat-focus-indicator"
              matripple=""
            />
            <div
              class="mat-step-icon mat-step-icon-state-number"
            >
              <div
                class="mat-step-icon-content"
              >
                <span
                  aria-hidden="true"
                >
                  2
                </span>
              </div>
            </div>
            <div
              class="mat-step-label"
            >
              <div
                class="mat-step-text-label"
              >
                Insert Date
              </div>
            </div>
          </mat-step-header>
          <div
            class="mat-vertical-content-container mat-stepper-vertical-line"
            inert=""
          >
            <div
              class="mat-vertical-stepper-content"
              role="tabpanel"
            >
              <div
                class="mat-vertical-content"
              >
                <form
                  class="ng-untouched ng-pristine ng-invalid"
                  novalidate=""
                >
                  <mat-form-field
                    class="mat-mdc-form-field pe-1 pt-3 mat-mdc-form-field-type-mat-input mat-mdc-form-field-has-icon-suffix mat-form-field-appearance-fill mat-primary ng-untouched ng-pristine ng-invalid"
                    style="width: 75%;"
                  >
                    <div
                      class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled mdc-text-field--no-label"
                    >
                      <div
                        class="mat-mdc-form-field-focus-overlay"
                      />
                      <div
                        class="mat-mdc-form-field-flex"
                      >
                        <div
                          class="mat-mdc-form-field-infix"
                        >
                          <input
                            aria-haspopup="dialog"
                            aria-required="true"
                            class="mat-mdc-input-element mat-datepicker-input ng-untouched ng-pristine ng-invalid mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
                            data-mat-calendar="mat-datepicker-a1"
                            formcontrolname="date"
                            id="mat-input-a4"
                            matinput=""
                            required=""
                          />
                          <mat-datepicker />
                        </div>
                        <div
                          class="mat-mdc-form-field-icon-suffix"
                        >
                          <mat-datepicker-toggle
                            class="mat-datepicker-toggle"
                            data-mat-calendar="mat-datepicker-a1"
                            maticonsuffix=""
                          >
                            <button
                              aria-expanded="false"
                              aria-haspopup="dialog"
                              aria-label="Open calendar"
                              class="mdc-icon-button mat-mdc-icon-button mat-mdc-button-base mat-unthemed _mat-animation-noopable"
                              mat-ripple-loader-centered=""
                              mat-ripple-loader-class-name="mat-mdc-button-ripple"
                              mat-ripple-loader-uninitialized=""
                              maticonbutton=""
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
                              />
                              <svg
                                aria-hidden="true"
                                class="mat-datepicker-toggle-default-icon"
                                fill="currentColor"
                                focusable="false"
                                height="24px"
                                viewBox="0 0 24 24"
                                width="24px"
                              >
                                <path
                                  d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                                />
                              </svg>
                              <span
                                class="mat-focus-indicator"
                              />
                              <span
                                class="mat-mdc-button-touch-target"
                              />
                            </button>
                          </mat-datepicker-toggle>
                        </div>
                      </div>
                      <div
                        class="mdc-line-ripple mdc-line-ripple--deactivating"
                        matformfieldlineripple=""
                      />
                    </div>
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
                    >
                      <div
                        class="mat-mdc-form-field-hint-wrapper"
                      >
                        <div
                          class="mat-mdc-form-field-hint-spacer"
                        />
                      </div>
                    </div>
                  </mat-form-field>
                  <div
                    class="pt-2"
                  >
                    <button
                      class="mdc-button mat-mdc-button-base mat-stepper-next mdc-button--raised mat-mdc-raised-button mat-unthemed _mat-animation-noopable"
                      mat-button=""
                      mat-raised-button=""
                      mat-ripple-loader-class-name="mat-mdc-button-ripple"
                      mat-ripple-loader-uninitialized=""
                      matsteppernext=""
                      type="submit"
                    >
                      <span
                        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        Next
                      </span>
                      <span
                        class="mat-focus-indicator"
                      />
                      <span
                        class="mat-mdc-button-touch-target"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mat-step"
        >
          <mat-step-header
            aria-disabled="true"
            aria-posinset="3"
            aria-selected="false"
            aria-setsize="4"
            class="mat-step-header mat-vertical-stepper-header mat-primary"
            role="tab"
            tabindex="-1"
          >
            <div
              class="mat-ripple mat-step-header-ripple mat-focus-indicator"
              matripple=""
            />
            <div
              class="mat-step-icon mat-step-icon-state-number"
            >
              <div
                class="mat-step-icon-content"
              >
                <span
                  aria-hidden="true"
                >
                  3
                </span>
              </div>
            </div>
            <div
              class="mat-step-label"
            >
              <div
                class="mat-step-text-label"
              >
                Add description (optional)
              </div>
            </div>
          </mat-step-header>
          <div
            class="mat-vertical-content-container mat-stepper-vertical-line"
            inert=""
          >
            <div
              class="mat-vertical-stepper-content"
              role="tabpanel"
            >
              <div
                class="mat-vertical-content"
              >
                <form
                  class="ng-untouched ng-pristine ng-invalid"
                  novalidate=""
                >
                  <mat-form-field
                    class="mat-mdc-form-field pt-1 mat-mdc-form-field-type-mat-input mat-form-field-appearance-fill mat-primary ng-untouched ng-pristine ng-invalid"
                  >
                    <div
                      class="mat-mdc-text-field-wrapper mdc-text-field mdc-text-field--filled mdc-text-field--no-label"
                    >
                      <div
                        class="mat-mdc-form-field-focus-overlay"
                      />
                      <div
                        class="mat-mdc-form-field-flex"
                      >
                        <div
                          class="mat-mdc-form-field-infix"
                        >
                          <input
                            aria-required="true"
                            class="mat-mdc-input-element ng-untouched ng-pristine ng-invalid mat-mdc-form-field-input-control mdc-text-field__input cdk-text-field-autofill-monitored"
                            formcontrolname="description"
                            id="mat-input-a5"
                            matinput=""
                            placeholder="Enter description"
                            required=""
                          />
                        </div>
                      </div>
                      <div
                        class="mdc-line-ripple mdc-line-ripple--deactivating"
                        matformfieldlineripple=""
                      />
                    </div>
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      class="mat-mdc-form-field-subscript-wrapper mat-mdc-form-field-bottom-align"
                    >
                      <div
                        class="mat-mdc-form-field-hint-wrapper"
                      >
                        <div
                          class="mat-mdc-form-field-hint-spacer"
                        />
                      </div>
                    </div>
                  </mat-form-field>
                  <div>
                    <button
                      class="mdc-button mat-mdc-button-base mat-stepper-next mdc-button--raised mat-mdc-raised-button mat-unthemed _mat-animation-noopable"
                      mat-button=""
                      mat-raised-button=""
                      mat-ripple-loader-class-name="mat-mdc-button-ripple"
                      mat-ripple-loader-uninitialized=""
                      matsteppernext=""
                      type="submit"
                    >
                      <span
                        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                      />
                      <span
                        class="mdc-button__label"
                      >
                        next
                      </span>
                      <span
                        class="mat-focus-indicator"
                      />
                      <span
                        class="mat-mdc-button-touch-target"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mat-step"
        >
          <mat-step-header
            aria-disabled="true"
            aria-posinset="4"
            aria-selected="false"
            aria-setsize="4"
            class="mat-step-header mat-vertical-stepper-header mat-primary"
            role="tab"
            tabindex="-1"
          >
            <div
              class="mat-ripple mat-step-header-ripple mat-focus-indicator"
              matripple=""
            />
            <div
              class="mat-step-icon mat-step-icon-state-number"
            >
              <div
                class="mat-step-icon-content"
              >
                <span
                  aria-hidden="true"
                >
                  4
                </span>
              </div>
            </div>
            <div
              class="mat-step-label"
            >
              <div
                class="mat-step-text-label"
              >
                Done
              </div>
            </div>
          </mat-step-header>
          <div
            class="mat-vertical-content-container"
            inert=""
          >
            <div
              class="mat-vertical-stepper-content"
              role="tabpanel"
            >
              <div
                class="mat-vertical-content"
              >
                <div
                  class="pt-2"
                >
                  <button
                    class="mdc-button mat-mdc-button-base me-1 mat-mdc-button mat-unthemed _mat-animation-noopable"
                    mat-button="tonal"
                    mat-ripple-loader-class-name="mat-mdc-button-ripple"
                    mat-ripple-loader-uninitialized=""
                  >
                    <span
                      class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                    />
                    <span
                      class="mdc-button__label"
                    >
                       Reset 
                    </span>
                    <span
                      class="mat-focus-indicator"
                    />
                    <span
                      class="mat-mdc-button-touch-target"
                    />
                  </button>
                  <button
                    class="mdc-button mat-mdc-button-base mdc-button--raised mat-mdc-raised-button mat-unthemed _mat-animation-noopable"
                    mat-button=""
                    mat-raised-button=""
                    mat-ripple-loader-class-name="mat-mdc-button-ripple"
                    mat-ripple-loader-uninitialized=""
                  >
                    <span
                      class="mat-mdc-button-persistent-ripple mdc-button__ripple"
                    />
                    <span
                      class="mdc-button__label"
                    >
                      Submit
                    </span>
                    <span
                      class="mat-focus-indicator"
                    />
                    <span
                      class="mat-mdc-button-touch-target"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </mat-stepper>
    </app-add-entry>
  </mat-dialog-content>
</add-entry-dialog>
`;
